api/v1/configurationpolicy_types.go

```diff
@@ -65,10 +65,15 @@ type Condition struct {
 	Message string `json:"message,omitempty" protobuf:"bytes,5,opt,name=message"`
 }
 
-// Target defines the list of namespaces to include/exclude
-type Target struct {
+// NamespaceSelector defines the list of namespaces to include/exclude
+type NamespaceSelector struct {
+	// +kubebuilder:validation:MinItems=1
 	Include []NonEmptyString `json:"include,omitempty"`
+	// +kubebuilder:validation:MinItems=1
 	Exclude []NonEmptyString `json:"exclude,omitempty"`
```

Comment:
Correct. It's the kind of thing where "if it's provided, put something in there". I was trying to make it more transparent that an empty array here would be rejected outright.

Now that you mention it though, I do need to explore `omitempty` a bit since I'm wondering if `matchLabels` is provided but empty might not show up, which could be confusing as a user looking back at the object.

Reply:
Okay, I've poked at it just in case, and providing `matchLabels: {}` and `matchExpressions: []` does still appear in the object. Setting `matchLabels: null` deletes the key.